pkg/commands/git.go

```diff
@@ -143,13 +134,15 @@ func (c *GitCommand) GetStatusFiles() []File {
 		stagedChange := change[0:1]
 		unstagedChange := statusString[1:2]
 		filename := c.OSCommand.Unquote(statusString[3:])
-		tracked := !includes([]string{"??", "A ", "AM"}, change)
+		_, untracked := map[string]bool{"??": true, "A ": true, "AM": true}[change]
+		_, hasUnstagedChanges := map[string]bool{" ": true, "U": true, "?": true}[stagedChange]
```

Comment:
HasStagedChanges and HasUnstagedChanges are in fact not mutually exclusive. For example if you have `git add`'d a file and then you make more changes, it both has staged changes and unstaged changes. So naming this variable 'hasUnstagedChanges' is not correct. I would add the exclamation mark in this line rather than below, i.e.

```
		_, hasStagedChanges := !map[string]bool{" ": true, "U": true, "?": true}[stagedChange]
```
or perhaps rename the variable to 'DoesNotHaveStagedChanges' (though I prefer the former)

Reply:
You can't add an exclamation mark like you did, you get two values